controlplane/rosa/controllers/rosacontrolplane_controller.go

```diff
@@ -344,6 +237,114 @@ func (r *ROSAControlPlaneReconciler) reconcileNormal(ctx context.Context, rosaSc
 	return ctrl.Result{}, nil
 }
 
+func ocmCluster(controlPlane *rosacontrolplanev1.ROSAControlPlane, now func() time.Time) (*clustersmgmtv1.Cluster, error) {
+	if now == nil {
+		now = time.Now
+	}
+	return clustersmgmtv1.NewCluster().
+		Name(controlPlane.Spec.RosaClusterName).
+		MultiAZ(true).
+		Product(
+			clustersmgmtv1.NewProduct().
+				ID("rosa"),
+		).
+		Region(
+			clustersmgmtv1.NewCloudRegion().
+				ID(*controlPlane.Spec.Region),
+		).
+		FIPS(false).
+		EtcdEncryption(false).
+		DisableUserWorkloadMonitoring(true).
+		Version(
+			clustersmgmtv1.NewVersion().
+				ID(*controlPlane.Spec.Version).
+				ChannelGroup("stable"),
+		).
+		ExpirationTimestamp(now().Add(1 * time.Hour)).
+		Hypershift(clustersmgmtv1.NewHypershift().Enabled(true)).
+		Network(
+			clustersmgmtv1.NewNetwork().
+				Type("OVNKubernetes").
+				MachineCIDR(*controlPlane.Spec.MachineCIDR),
+		).
+		AWS(
+			clustersmgmtv1.NewAWS().
+				AccountID(*controlPlane.Spec.AccountID).
+				BillingAccountID(*controlPlane.Spec.AccountID).
+				SubnetIDs(controlPlane.Spec.Subnets...).
+				PrivateLink(controlPlane.Spec.AWS.PrivateLink).
```

Comment:
if `PrivateLink` is enabled we should set the API server listening mode to `Internal`
see https://github.com/openshift/rosa/blob/4639f9824693ae52d013716a2b6b47758180eaea/pkg/ocm/clusters.go#L1041-L1049